frontend/server/src/Controllers/User.php

```diff
@@ -1579,18 +1552,19 @@ public static function apiSelectCoderOfTheMonth(\OmegaUp\Request $r): array {
             \OmegaUp\DAO\DAO::transBegin();
             foreach ($users as $index => $user) {
                 $newCoderOfTheMonth = new \OmegaUp\DAO\VO\CoderOfTheMonth([
+                    'coder_of_the_month_id' => $user['coder_of_the_month_id'],
                     'user_id' => $user['user_id'],
                     'school_id' => $user['school_id'],
-                    'time' => $dateToSelect,
-                    'rank' => $index + 1,
-                    'category' => $category,
+                    'time' => $user['time'],
+                    'rank' => $user['rank'],
+                    'category' => $user['category'],
                 ]);
                 // All users calculated as CoderOfTheMonth are going to be saved on database,
                 // the one selected by the mentor is gonna have the field 'selected_by' filled.
                 if ($user['username'] === $r['username']) {
                     $newCoderOfTheMonth->selected_by = $r->identity->identity_id;
                 }
-                \OmegaUp\DAO\CoderOfTheMonth::create($newCoderOfTheMonth);
+                \OmegaUp\DAO\CoderOfTheMonth::update($newCoderOfTheMonth);
```

Comment:
esta línea sólo se debería ejecutar si L1564 es verdadero. en otras palabras:

```suggestion
                // Only the CoderOfTheMonth selected by the mentor is going to be
                // updated.
                if ($user['username'] !== $r['username']) {
                    continue;
                }
                $newCoderOfTheMonth->selected_by = $r->identity->identity_id;
                \OmegaUp\DAO\CoderOfTheMonth::update($newCoderOfTheMonth);
```

Reply:
listo